gdal/doc/source/drivers/vector/sosi.rst

```diff
@@ -7,3 +7,8 @@ Norwegian SOSI Standard
 .. shortname:: SOSI
 
 This driver requires the FYBA library.
+
+ poDriver->SetMetadataItem( GDAL_DMD_OPENOPTIONLIST,
```

Comment:
No, this is code that should go to ogr/ogrsf_frmts/sosi/ogrsosidriver.cpp

In the rst file, you must add reStructured text. See for example doc/source/drivers/vector/csv.rst which has an "Open Options" section